main/lsp/LSPTypechecker.h

```diff
@@ -38,11 +38,28 @@ class UndoState;
 class LSPTypechecker final {
     /** Contains the ID of the thread responsible for typechecking. */
     std::thread::id typecheckerThreadId;
-    /** GlobalState used for typechecking. */
+    /**
+     * GlobalState used for typechecking. It is replaced during indexing on the slow path, and is always replaced with
+     * a GlobalState that derives from the one created during LSP initialization. This derivation is guaranteed by
+     * passing a copy of that GlobalState to the indexer thread at the end of initialization, and then copying that
+     * GlobalState back over to the typechecker to use as the starting point for the next slow path.
+     */
     std::unique_ptr<core::GlobalState> gs;
-    /** Trees that have been indexed (with initialGS) and can be reused between different runs */
+    /**
+     * Trees that have been indexed with the GlobalState that was originally supplied during initialization
+     * (initialGS). As all values of this->gs will derive from that initial GlobalState, none of these trees will be
+     * re-indexed for subsequent slow path runs. An additional consequence of this->gs deriving from the initialization
+     * value of GlobalState is that they will continue to be valid when used with this->gs (see the comment on this->gs
+     * for an explanation of how this derivation is ensured). Finally, this vector will never be updated after the
+     * indexing done during initialization, and represents a snapshot of the workspace when LSP was started.
```

Comment:
Is this actually true? We never update `indexed`? Could it be `const`?

I'm just kind of shocked that this isn't updated more than once.

Reply:
Not true, I fixed this comment soon after in #8527 😅